Userland/Applications/KeyboardSettings/KeyboardSettingsWidget.cpp

```diff
@@ -49,11 +55,17 @@ class KeymapSelectionDialog final : public GUI::Dialog {
     ByteString selected_keymap() { return m_selected_keymap; }
 
 private:
-    KeymapSelectionDialog(Window* parent_window, Vector<ByteString> const& selected_keymaps)
+    static ErrorOr<NonnullRefPtr<KeymapSelectionDialog>> create(Window* parent_window, Vector<ByteString> const& selected_keymaps)
+    {
+        auto widget = TRY(KeyboardSettings::KeymapDialog::try_create());
+        auto dialog = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) KeymapSelectionDialog(parent_window, selected_keymaps, widget)));
+        return dialog;
```

Comment:
This looks like a use-after-free on `widget`. `KeymapDialog` should probably be passed as a `NonnullRefPtr`.